class-two-factor-core.php

```diff
@@ -745,11 +753,12 @@ protected static function hash_login_nonce( $nonce ) {
 	 * @since 0.1-dev
 	 *
 	 * @param int $user_id User ID.
-	 * @return array
+	 * @return array|false
 	 */
 	public static function create_login_nonce( $user_id ) {
 		$login_nonce = array(
-			'expiration' => time() + HOUR_IN_SECONDS,
+			'user_id'    => $user_id,
+			'expiration' => time() + ( 10 * MINUTE_IN_SECONDS ),
```

Comment:
I'm wondering if we should store the creation time here and verify the expiration at the login state? This would allow adjusting the expiration logic based on certain conditionals via filters, for example.

Reply:
🤔 I don't have a strong opinion, except that that might be a bit outside the scope of this PR, and there're a lot of other things to do for `0.8`. Would it be better to open a ticket for that as a future iteration?

Alternatively, something quick like this might cover the use cases you were thinking of, and be forwards-compatible for future expansion:

```php
function __construct() {
	self::LOGIN_NONCE_EXPIRATION = apply_filters( 'two_factor_login_nonce_expiration', 10 * MINUTE_IN_SECONDS );
}

public static function create_login_nonce( $user_id ) {
	$login_nonce = array(
		'user_id'    => $user_id,
		'expiration' => time() + self::LOGIN_NONCE_EXPIRATION,
	);
}
```

I'm open to either, though. Do you have a preference?

Reply:
Core uses `expiration` in the user auth cookies, in preference for created + expiration being filtered after-the-fact.

Reply:
OK, let's leave it as is for this changeset.